platform/pwl.c

```diff
@@ -1358,8 +1375,13 @@ pwl_display_destroy (PwlDisplay *self)
     pwl_display_egl_deinit (self);
 
     if (self->display) {
-        display_stop_keyboard_repeat (self);
+        PwlSeat *item, *tmp;
 
+        wl_list_for_each_safe (item, tmp, &self->seats, link) {
```

Comment:
Here it's fine to use the `wl_list_for_each()` macro. You only need to use `wl_list_for_each_safe()` if you are planning to modify the list while iterating over it.

Reply:
Code simplified and encapsulated in the `pwl_seat_destroy()`